docs/api/v3.rst

```diff
@@ -464,6 +464,7 @@ Version detail
             },
             "urls": {
                 "documentation": "https://pip.pypa.io/en/stable/",
+                "edit": "https://readthedocs.org/dashboard/pip/version/stable/edit/",
```

Comment:
Maybe worth to have a different field, `dashboard`, for all the URLs under our own platform instead of mixing them with other non-related URLs that go to external sites?

* `urls.dashboard.edit`
* `dashboard.edit`

Reply:
Are we expecting that many URLs in `urls` that we need the disambiguation or further nesting? It doesn't seem like API returns are going to have more the 4 or 5 urls at a time -- documentation links for some, perhaps CRUD view URLs in addition? I always found it worked well as a flat map. The implementation seems easy to understand and easy to use, and it also matches other providers' APIs.

Reply:
Coming back to this discussion, do we have any other API returns that we're sending back dashboard links? This could be a pattern we see more as more of our expensive interactions get pushed through JS, but this is hard for me to estimate right now.

If `urls` is a common field in API return, and it is mostly used for external site urls, then perhaps that is a good reason to nest dashboard links. I'm feeling maybe -0 on the change though, but I also don't have a good grasp on how we're using external links through the rest of the API currently. This feels like a pattern we wouldn't be using a lot, but I'm not certain of this.

Reply:
Also, if we want to really expand usage of dashboard links in our URL, that could maybe push us towards a dedicated structure. For example, this is a single case that we need for the dashboard, but maybe it's helpful to include any dashboard link we can in the API response. I think GitHub or Stripe does something similar, but they perhaps also don't have a tertiary data structure for external site URLS on top of API URLs and dashboard URLs.

Reply:
I always prefer nested/dictionary fields in those cases where we don't _exactly_ know how our API is going to be used and when we know that it's not _100% finished_ and we may accept/add more fields. Using dictionaries allow us to add these "future" fields without breaking the structure and keeping the semantic of the field name consistent over time. That's why I was suggesting to split dashboard and external URLs.

This example sounds good to me: https://readthedocs.org/api/v3/projects/docs/versions/stable/

```js
    "urls": {
        "dashboard": {
            "edit": "https://readthedocs.org/dashboard/pip/version/stable/edit/"
        },
        "documentation": "https://docs.readthedocs.io/en/stable/",
        "vcs": "https://github.com/readthedocs/readthedocs.org/tree/5.14.3/"
    },
```

Maybe we can't think about more dashboard URLs for the `Version` object, but we may have more for `Project` (e.g. admin, edit, versions, etc) and I think it makes sense to have the same structure for all the resources.